metrics-operator/controllers/metrics/controller.go

```diff
@@ -125,6 +115,42 @@ func (r *KeptnMetricReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return reconcile, err
 }
 
+func (r *KeptnMetricReconciler) updateMetric(metric *metricsapi.KeptnMetric, value string, rawValue []byte, reconcile ctrl.Result, err error) (*metricsapi.KeptnMetric, ctrl.Result) {
+	if metric.Spec.Range != nil && metric.Spec.Range.StoredResults > 0 {
+		intervalResult := metricsapi.IntervalResult{
+			LastUpdated: metav1.Time{Time: time.Now().UTC()},
+			ErrMsg:      "",
+		}
+
+		if err != nil {
+			r.Log.Error(err, "Failed to evaluate the query", "Response from provider was:", (string)(rawValue))
+			intervalResult.ErrMsg = err.Error()
+		} else {
+			intervalResult.Value = value
+			intervalResult.Range = metric.Spec.Range
+		}
+
+		if len(metric.Status.IntervalResults) >= int(metric.Spec.Range.StoredResults) {
+			metric.Status.IntervalResults = append(metric.Status.IntervalResults[1:], intervalResult)
+		} else {
+			metric.Status.IntervalResults = append(metric.Status.IntervalResults, intervalResult)
+		}
+	} else {
+		if err != nil {
+			r.Log.Error(err, "Failed to evaluate the query", "Response from provider was:", (string)(rawValue))
+			reconcile = ctrl.Result{Requeue: false}
+			metric.Status.ErrMsg = err.Error()
+		} else {
+			metric.Status.Value = value
```

Comment:
one small improvement we can do to the previous logic is to also set `status.ErrMsg` to an empty string here in case there was an error in a previous reconciliation attempt

Reply:
Fixed